connector/connect/server/src/main/scala/org/apache/spark/sql/connect/utils/ErrorUtils.scala

```diff
@@ -115,6 +115,17 @@ private[connect] object ErrorUtils extends Logging {
           if (sparkThrowable.getErrorClass != null) {
             sparkThrowableBuilder.setErrorClass(sparkThrowable.getErrorClass)
           }
+          for (queryCtx <- sparkThrowable.getQueryContext) {
+            sparkThrowableBuilder.addQueryContexts(
+              FetchErrorDetailsResponse.QueryContext
+                .newBuilder()
+                .setObjectType(queryCtx.objectType())
+                .setObjectName(queryCtx.objectName())
+                .setStartIndex(queryCtx.startIndex())
```

Comment:
Hm, this should actually fail after https://github.com/apache/spark/pull/43334 PR because `DataFrameQueryContext` throws an exception now (https://github.com/apache/spark/pull/43334/files#diff-b3bc05fec45cd951053b2876c71c7730b63789cb4336a7537a6654c724db3241R586-R589).

Seems like we miss this information Spark Connect client sides for now so it seems working .. but we should carry this context for DataFrame operations ..

Reply:
@MaxGekk I think there's a bug not only here but without Spark Connect.

```scala
scala> try {
     |   spark.range(1).select("a")
     | } catch {
     |     case e: org.apache.spark.sql.catalyst.ExtendedAnalysisException => println(e.getQueryContext)
     | }
[Lorg.apache.spark.QueryContext;@6a9d7514
val res2: Any = ()
```
It doesn't contain `QueryContext` from https://github.com/apache/spark/pull/43334 ..

Reply:
Alright, there are two issues. TL;DR

1. There are many places that does not provide `Origin` that contains `QueryContext` at `QueryCompilationErrors`. So the `QueryContext` is often missing (e.g., the case above). cc @MaxGekk 

2. `withOrigin` is not being invoked, and the stacktrace is undefined in Spark Connect server. So always we set `SQLQueryContext`. That's why this specific code did not throw an exception from https://github.com/apache/spark/pull/43334/files#diff-b3bc05fec45cd951053b2876c71c7730b63789cb4336a7537a6654c724db3241R586-R589 because it has never been `DataFrameContext`. We should invoke `withOrigin` in Spark Connect server. cc @heyihong @juliuszsompolski FYI.

3. The current logic in `ErrorUtils.scala` should handle the case of `DataFrameQueryContext` e.g., `DataFrameQueryContext.stopIndex()` will throw an exception. Should we:
  - Set a default value in `DataFrameQueryContext.stopIndex()` instead of throwing an exception? @MaxGekk 
  - Or, make the protobuf message for this optional, and throw an exception from Spark Connect client sides? @heyihong @juliuszsompolski

Reply:
> We should invoke withOrigin in Spark Connect server.

do you mean client? Server side stacktrace is not interesting to users to understand their own code mistakes.

Reply:
```
case class Origin(
    line: Option[Int] = None,
    startPosition: Option[Int] = None,
    startIndex: Option[Int] = None,
    stopIndex: Option[Int] = None,
    sqlText: Option[String] = None,
    objectType: Option[String] = None,
    objectName: Option[String] = None,
    stackTrace: Option[Array[StackTraceElement]] = None) {

  lazy val context: QueryContext = if (stackTrace.isDefined) {
    DataFrameQueryContext(stackTrace.get.toImmutableArraySeq)
  } else {
    SQLQueryContext(
      line, startPosition, startIndex, stopIndex, sqlText, objectType, objectName)
  }
```

So it's considered as `SQLQueryContext` .. even when you're using `DataFrame` API in Spark Connect client.

Reply:
Either setting a default value or making the protobuf field optional should work. It depends on the semantic of `DataFrameQueryContext.stopIndex()`.

cc: @MaxGekk

Reply:
Let's make sure we fix this.